locator/tablets.hh

```diff
@@ -148,6 +139,12 @@ public:
         return tablet_id(size_t(t) + 1);
     }
 
+    /// Returns shard id which is a replica for a given tablet on a given host.
+    /// If there is no replica on a given host, returns nullopt.
+    /// If the topology is transitional, also considers the new replica set.
+    /// The old replica set is preferred in case of ambiguity.
+    std::optional<shard_id> get_shard(tablet_id, host_id) const;
```

Comment:
Shouldn't the direction matter? reads shouldn't consider the new node during write_both_read_old.

Reply:
Yes, transitional topology is not really handled properly yet anywhere.